funcs_utils.php

```diff
@@ -287,15 +287,23 @@ function current_branch_cms_major(
         return MetaData::HIGHEST_STABLE_CMS_MAJOR;
     }
 
+    $contents = '';
     if ($composerJson) {
         $contents = $composerJson;
     } elseif (check_file_exists('composer.json')) {
         $contents = read_file('composer.json');
     }
     $composerJson = json_decode($contents);
     if (is_null($composerJson)) {
-        $lastError = json_last_error();
-        error("Could not parse from composer.json - last error was $lastError");
+        // allow composer.json to be missing for "misc" repos
+        // except for those repos that do have composer.json file
+        if (!is_misc() || in_array($GITHUB_REF, [
+            'silverstripe/silverstripe-frameworktest',
+            'silverstripe/silverstripe-module'
+        ])) {
```

Comment:
```suggestion
        if (!is_misc() && !check_file_exists('composer.json')) {
```
We don't need to keep a list of all misc modules that have composer.json files - it either has the file or it doesn't.

Reply:
Updated